spec/cohortManager.spec.js

```diff
@@ -0,0 +1,275 @@
+const CohortManager = require('../src/cohortManager.js')
+
+describe('Cohort Manager', () => {
+  let app
+
+  beforeEach(() => {
+    app = new CohortManager()
+  })
+
+  it('creates a new cohort', () => {
+    const expected = {
+      id: 1,
+      cohortName: 'cohort10',
+      students: []
+    }
+    const result = app.addCohort('cohort10')
+    expect(result).toEqual(expected)
+  })
+  it('try to add a new cohort without a name', () => {
+    const result = () => app.addCohort('')
+    expect(result).toThrowError()
+  })
+  it('cohort manager has no cohorts', () => {
+    const expected = []
+    const result = app.cohorts
+    expect(result).toEqual(expected)
+  })
+
+  it('search for a specific cohort', () => {
+    app.addCohort('cohort10')
+    app.addCohort('cohort06')
+    app.addCohort('cohort07')
+    const result = app.searchByCohortName('cohort10')
+    const expected = {
+      id: 1,
+      cohortName: 'cohort10',
+      students: []
+    }
+    expect(result).toEqual(expected)
+  })
+  it('error message when cohort name does not exist', () => {
+    app.addCohort('cohort05')
+    app.addCohort('cohort06')
+    app.addCohort('cohort07')
+    const result = () => app.searchByCohortName('cohort10')
+    expect(result).toThrowError('Cohort was not found')
+  })
+  it('add a new student to a cohort - testing 2 classes', () => {
+    app.addCohort('cohort05')
+    const expected = {
+      id: 1,
+      cohortName: 'cohort05',
+      students: [
+        {
+          firstName: 'carolina',
+          lastName: 'arruda',
+          githubUsername: 'carolarruda',
+          email: 'carolinacarruda@sapo.pt'
+        }
+      ]
+    }
+    const result = app.addStudent(
+      'cohort05',
+      'carolina',
+      'arruda',
+      'carolarruda',
+      'carolinacarruda@sapo.pt'
+    )
+
+    for (let i = 0; i < result.length; i++) {
+      expect(result[i].id).toEqual(expected[i].id)
+      expect(result[i].cohortName).toEqual(expected[i].cohortName)
+      expect(result[i].students[0].firstName).toEqual(
+        expected[i].students[0].firstName
+      )
+      expect(result[i].students[0].lastName).toEqual(
+        expected[i].students[0].lastName
+      )
+      expect(result[i].students[0].githubUsername).toEqual(
+        expected[i].students[0].githubUsername
+      )
+      expect(result[i].students[0].email).toEqual(expected[i].students[0].email)
+    }
+
+    expect(result.students[0].email).toEqual(expected.students[0].email)
```

Comment:
I like how you've laid out multiple expects for the same test! Nice way to tidy up the code!

Reply:
This was a workaround for the unpredictability of my Ids tbh